pkg/apis/pipeline/v1beta1/task_validation.go

```diff
@@ -62,6 +62,7 @@ func (ts *TaskSpec) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 	errs = errs.Also(validateSteps(ctx, mergedSteps).ViaField("steps"))
 	errs = errs.Also(ts.Resources.Validate(ctx).ViaField("resources"))
+	errs = errs.Also(validateParameterNames(ts.Params).ViaField("params"))
 	errs = errs.Also(ValidateParameterTypes(ts.Params).ViaField("params"))
```

Comment:
nit: we could have a `ValidateParamSpecs` function that does both of those at the same time. Probably an opmitization at that point of time though.